libsql-wal/src/registry.rs

```diff
@@ -126,9 +128,8 @@ where
                     update_durable(fno, notifier, durable_frame_no, namespace).await;
                 })
             });
-            self.storage
-                .store(&shared.namespace, sealed.clone(), None, cb);
-            new.tail().push(sealed);
+            new.tail().push(sealed.clone());
+            self.storage.store(&shared.namespace, sealed, None, cb);
```

Comment:
was there a bug here? Curious why these got switched

Reply:
Yes i explained in the commit message. I have a storage implementation that's so fast it tried to checkpoint before we added the sealed sediment to the list. Swapping the two so that the check pointer always finds a segment of it's notified